src/filter_base.cpp

```diff
@@ -124,6 +124,11 @@ namespace RobotLocalization
   {
   }
 
+  void FilterBase::reset()
+  {
+    initialized_ = false;
```

Comment:
I notice this doesn't reset any of the state variables. This class should ideally work independently of the `RosFilter` class, so I would think a reset should reset all the member variables. Perhaps we can move all of the constructor logic into this method and then call `reset()` from the constructor?